pkg/instance/service.go

```diff
@@ -661,6 +664,79 @@ func (s service) FindInstances(user *model.User, presets bool) ([]GroupsWithInst
 	return instances, err
 }
 
+type InstanceStatus string
+
+const (
+	NotDeployed        InstanceStatus = "NotDeployed"
+	Pending            InstanceStatus = "Pending"
+	Booting            InstanceStatus = "Booting"
+	BootingWithRestart InstanceStatus = "Booting (%d)"
+	Running            InstanceStatus = "Running"
+	Error              InstanceStatus = "Error"
+)
+
+func (s service) GetStatus(instance *model.Instance) (InstanceStatus, error) {
+	ks, err := NewKubernetesService(instance.Group.ClusterConfiguration)
+	if err != nil {
+		return "", err
+	}
+
+	pod, err := ks.getPod(instance, "")
+	if err != nil {
+		if strings.HasPrefix(err.Error(), "no pod found using the selector: ") {
+			return NotDeployed, nil
+		}
+		return "", err
+	}
+
+	if pod.Status.Phase == v1.PodPending {
```

Comment:
Maybe use a `switch` statement instead, as we're checking the pod status for 3 different cases (and might have more?).

Reply:
I updated the code but I'm not sure I find it more readable. Please take another look and tell me which version you like the most :smile:

Reply:
I suppose readability is the same with both options (at least for me). Choose whichever you prefer 👍